code/modules/food_and_drinks/machinery/doner_machine.dm

```diff
@@ -101,110 +122,108 @@
 	return ..()
 
+/obj/item/doner_stick/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	. = ..()
+	if(raw && meat_level < 5 && meat_types[held_item.icon_state] && istype(held_item, /obj/item/food/meat/slab))
+		context[SCREENTIP_CONTEXT_LMB] = "Add meat"
+		return CONTEXTUAL_SCREENTIP_SET
+	else if(!raw && held_item.tool_behaviour == TOOL_KNIFE)
+		context[SCREENTIP_CONTEXT_LMB] = "Slice meat"
+		return CONTEXTUAL_SCREENTIP_SET
+	return NONE
+
 /obj/item/doner_stick/attackby(obj/item/item, mob/user, params)
 	if(istype(item, /obj/item/food/meat/slab))
-		var/obj/item/food/meat/slab/meatslab = item
-		if(!meat_types[meatslab.icon_state])
-			to_chat(user, span_warning("You cant put this meat!"))
-			return
+		var/obj/item/food/meat/slab/slab = item
 		if(!raw)
-			to_chat(user, span_warning("You cant put a raw meat to cooked doner!"))
+			to_chat(user, span_warning("You cannot add more meat after [src] is cooked!"))
+			return
+		if(meat_level == 5)
+			to_chat(user, span_warning("You cannot add more meat to it."))
 			return
-		if(donericon && donericon != meat_types[meatslab.icon_state])
-			to_chat(user, span_warning("You cant put this type of meat to [donericon] doner!"))
+		if(!isnull(meat_type) && meat_type != meat_types[slab.icon_state])
+			to_chat(user, span_warning("You cannot mix [slab] with [meat_type] doner!"))
 			return
-		if(meatnum == 5)
-			to_chat(user, span_warning("You cant add more meat to it."))
+		if(!meat_types[slab.icon_state])
+			to_chat(user, span_warning("You cannot use this meat!"))
 			return
 		if(!reagents)
 			create_reagents(250, INJECTABLE)
-		meatslab.reagents.trans_to(src, meatslab.reagents.total_volume)
-		for(var/taste in meatslab.tastes)
-			tastes += list(taste = meatslab.tastes[taste])
-		donericon = meat_types[meatslab.icon_state]
-		meatnum++
-		qdel(meatslab)
+		slab.reagents.trans_to(src, slab.reagents.total_volume)
+		if(!meat_type)
+			meat_type = meat_types[slab.icon_state]
+		for(var/taste in slab.tastes)
+			var/taste_stick = tastes[taste]
+			var/taste_slab = slab.tastes[taste]
+			if(isnull(taste_stick) || taste_stick < taste_slab)
+				tastes[taste] = taste_slab
+		meat_level++
+		qdel(slab)
 		update_appearance()
 	else if(item.tool_behaviour == TOOL_KNIFE)
-		if(!meatnum)
-			to_chat(user, span_warning("There is no meat to cut!"))
+		if(!meat_level)
+			to_chat(user, span_warning("You cannot cut steel!"))
 			return
 		if(raw)
-			to_chat(user, span_warning("This doner is raw!"))
+			to_chat(user, span_warning("[src] is raw!"))
 			return
-		var/donertype = (donericon == "meat") ? /obj/item/food/doner/yaprak/et : /obj/item/food/doner/yaprak/tavuk
-		var/volume = (reagents.total_volume / (meatnum * 2))
-		var/turf/drop_loc = drop_location()
+		var/doner_type = (meat_type == "meat") ? /obj/item/food/doner/yaprak/et : /obj/item/food/doner/yaprak/tavuk
+		var/volume = (reagents.total_volume / (meat_level * 2)) // bunun hesaplamasını anlamadım
```

Comment:
😁

Reply:
o volume olayı tüm parçalara eşit reagent dağıtılsın diye
mesela diyelimki total volume 100
1,2.ye 100/(5x2) = 10
3,4.ye 80/(4x2) = 10
5,6.ye 60/(3x2) = 10
7,8.ye 40/(2x2) = 10
9,10.ye 20/(1x2) = 10

Reply:
okey anladım commenti sil ama koddan

Reply:
tamam